### DISCUSSION TITLE
Par0 values

### DISCUSSION THREAD

**Discussion by birdwrangler** [bmcclintock/momentuHMM#55] [Q&A] [answered]
I’m working through the momentuHMM caracara tutorial and am wondering how the starting values for optimization (gamma means & SD) were determined? It is my understanding that each value is associated with one of the 4 (assumed) states, but I’ve not been able to determine how to calculate or estimate these values. 

I realize this may not be a simple answer, but any clarification is welcome!

`## starting values for optimization`
`# gamma means for states 1,...,N`
`mu <- c(0.005,0.01,0.05,0.1) `     
`# gamma standard deviations for states 1,...,N`                          
`sig <- c(0.01,0.02,0.03,0.06)`               
`Par0 <- list(VDBA = c(mu,sig))`

**Comment by bmcclintock** [comment, marked as answer]
Specifying "good" initial values can be challenging, especially with relatively small or noisy datasets where the model does not fit the data well. It can also involve a lot of trial-and-error, with the goal being to try and find initial values that yield the largest log-likelihood (or, equivalently, the smallest negative log-likelihood). The `retryFits` argument in `fitHMM` is intended to help with this process by iteratively re-fitting the model using random perturbations of the current parameter values. For general advice, I'd suggest the following:

https://cran.r-project.org/web/packages/moveHMM/vignettes/moveHMM-starting-values.pdf

See also the references in the "Model fitting" section of this [tutorial](https://onlinelibrary.wiley.com/action/downloadSupplement?doi=10.1111%2Fele.13610&file=ele13610-sup-0002-Supinfo.pdf). Although less commonly used, a grid search can also be effective:

https://blogs.sas.com/content/iml/2018/06/25/grid-search-for-parameters-sas.html

**Comment by birdwrangler** [comment]
Thank you for the quick response, this is super helpful!
